junit-platform-commons/src/main/java/org/junit/platform/commons/support/AnnotationSupport.java

```diff
@@ -153,6 +170,45 @@ public static boolean isAnnotated(AnnotatedElement element, Class<? extends Anno
 		return AnnotationUtils.findPublicAnnotatedFields(clazz, fieldType, annotationType);
 	}
 
+	/**
+	 * Find all {@code public} {@linkplain Field fields} values of the supplied instance
+	 * that are annotated or <em>meta-annotated</em> with the specified {@code annotationType}.
+	 *
+	 * <p>Consult the Javadoc for {@link Class#getFields()} for details on
+	 * inheritance and ordering.
+	 *
+	 * @param object the instance in which to find the fields; never {@code null}
+	 * @param annotationType the annotation type to search for; never {@code null}
+	 * @return the list of all such field values found; neither {@code null} nor mutable
+	 * @see Class#getFields()
+	 */
+	public static List<Object> findPublicAnnotatedFieldsValues(Object object,
+			Class<? extends Annotation> annotationType) {
+		return ReflectionUtils.getFieldValuesForInstance(findPublicAnnotatedFields(object.getClass(), annotationType),
+			object);
+	}
+
+	/**
+	 * Find all {@code public} {@linkplain Field fields} values of the supplied instance
+	 * that are of the specified {@code fieldType} and annotated
+	 * or <em>meta-annotated</em> with the specified {@code annotationType}.
+	 *
+	 * <p>Consult the Javadoc for {@link Class#getFields()} for details on
+	 * inheritance and ordering.
+	 *
+	 * @param object the instance in which to find the fields; never {@code null}
+	 * @param fieldType the type of field to find; never {@code null}
+	 * @param annotationType the annotation type to search for; never {@code null}
+	 * @return the list of all such field values found; neither {@code null} nor mutable
+	 * @see Class#getFields()
+	 */
+	public static <T> List<T> findPublicAnnotatedFieldsValues(Object object, Class<T> fieldType,
```

Comment:
Why limit the results to `public` fields?

Reply:
Should we limit the results to public fields or all fields ?

Reply:
See related answer.

Reply:
removed public from name